### PR TITLE
Fix .xy floating point number parsing in eval()

### PR DESCRIPTION
**Pull request by vpodzime**
The first commit is just paving the way. Only the changes in the *libpromises/math.peg* file are relevant for review.

merge together:
https://github.com/cfengine/core/pull/3778

### CODE REVIEW

File: 3rdparty/peg-0.1.15/src/peg.peg-c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* A recursive-descent parser generated by peg 0.1.13 */
+/* A recursive-descent parser generated by peg 0.1.15 */
```
Comment (nickanderson) [inline 3rdparty/peg-0.1.15/src/peg.peg-c]: so did this actually update the dependacny version? We should note that inthe dependancy tables in buildscripts I guess.

Comment (vpodzime) [reply 3rdparty/peg-0.1.15/src/peg.peg-c]: This is not a runtime dependency, not even a build dependency. It's only needed when you want to regenerate the `math.pc` file from the `math.peg` grammar (which we do manually).

And I only updated `peg` to a version that was clearly used before to (re)generate the `math.pc` file.

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2487/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2487/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2487/

File: 3rdparty/peg-0.1.15/src/peg.peg-c (hunk 1 above)
Comment (nickanderson) [reply 3rdparty/peg-0.1.15/src/peg.peg-c]: I see. thanks.

**Pull request merged**